lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -152,11 +152,13 @@
 <%- @domain_volumes.each_with_index do |volume, index| -%>
     <disk type='file' device='disk'>
       <alias name='ua-box-volume-<%= index -%>'/>
-      <driver name='qemu' type='qcow2' <%=
-        @disk_driver_opts.empty? ? "cache='#{volume[:cache]}'" :
-            @disk_driver_opts.reject { |k,v| v.nil? }
-                             .map { |k,v| "#{k}='#{v}'"}
-                             .join(' ') -%>/>
+      <driver name='qemu'<%=
+        '%{type}%{cache}%{io}%{copy_on_read}%{discard}%{detect_zeroes}' \
+          % { :type => 'qcow2', :cache => "#{volume[:cache]}", :io => nil, :copy_on_read => nil, :discard => nil, :detect_zeroes => nil }
+            .merge(@disk_driver_opts)
+            .select { |k,_| [:type, :cache, :io, :copy_on_read, :discard, :detect_zeroes].include? k }
+            .map { |k,v| v.nil? ? [k, ""] : [k, " #{k}='#{v}'"]}.to_h
```

Comment:
I didn't think you need to interpolate all of the results, think the following achieves the same without needing to specify all of the options either:
```
      <driver name='qemu' <%=
        {
            :type => 'qcow2',
            :cache => "#{volume[:cache]}",
         }.merge(@disk_driver_opts).
             reject { |k,v| v.nil? }.
             map { |k,v| "#{k}='#{v}'"}.
             join(' ') -%>/>
```

Reply:
The approach used in the PR was to specifically to avoid sorting issues that generated keys in a different order than the test artifacts. I think if the original `map()/join()` approach is used the test artifacts have to be modified instead. I'm perfectly happy with whichever you think is better.

Reply:
Think I'm managed to update in a way that avoid impacting the tests while also avoiding explicit references to the only config options that are valid.